src/daemon/daemon.cpp

```diff
@@ -335,6 +341,8 @@ bool Daemon::parseConfig(const QJsonObject& obj, InterfaceConfig& config) {
 bool Daemon::deactivate(bool emitSignals) {
   Q_ASSERT(wgutils() != nullptr);
 
+  m_state = Inactive;
```

Comment:
I am little bit worried about the accuracy of setting this state here. There are multiple paths here where deactivation will fail, when that happens does mean the VPN remain active? Or do we actually go into an error state? Would it make sense to introduce `Error` and `Deactivating` states? @oskirby